test/integration/remote/test_local_recipes_index.py

```diff
@@ -331,3 +331,37 @@ class Zlib(ConanFile):
         c.run(f"remote add local '{folder}'")
         c.run("install --requires=zlib/[*] --build missing", assert_error=True)
         assert "NameError: name 'ConanFile' is not defined" in c.out
+
+
+class TestPythonRequires:
+    @pytest.fixture(scope="class")
+    def c3i_pyrequires_folder(self):
+        folder = temp_folder()
+        recipes_folder = os.path.join(folder, "recipes")
+        config = textwrap.dedent("""
+            versions:
+              "1.0":
+                folder: all
+            """)
+        zlib = textwrap.dedent("""
+            from conan import ConanFile
+            class Zlib(ConanFile):
+                name = "pkg"
+                python_requires = "pyreq/1.0"
+                """)
+        save_files(recipes_folder,
+                   {"pkg/config.yml": config,
+                    "pkg/all/conanfile.py": zlib,
+                    "pyreq/config.yml": config,
+                    "pyreq/all/conanfile.py": str(GenConanfile("pyreq", "1.0"))})
+        return folder
+
+    def test_install(self, c3i_pyrequires_folder):
```

Comment:
If it's not too much, I'd add a test to explicitly state that a pyreq coming from a different remote is not supported? Or is it enought with the issue message?

Reply:
I think we need to consider the use cases with more detail, this is why this is a draft, lets talk about it.